runners/spark/src/main/java/org/apache/beam/runners/spark/translation/GroupCombineFunctions.java

```diff
@@ -146,20 +147,40 @@
     JavaPairRDD<K, WindowedValue<KV<K, InputT>>> inRddDuplicatedKeyPair =
         rdd.mapToPair(TranslationUtils.toPairByKeyInWindowedValue());
 
-    JavaPairRDD<K, SerializableAccumulator<KV<K, AccumT>>> accumulatedResult =
-        inRddDuplicatedKeyPair.combineByKey(
-            input ->
-                SerializableAccumulator.of(sparkCombineFn.createCombiner(input), iterAccumCoder),
-            (acc, input) ->
-                SerializableAccumulator.of(
-                    sparkCombineFn.mergeValue(input, acc.getOrDecode(iterAccumCoder)),
-                    iterAccumCoder),
-            (acc1, acc2) ->
-                SerializableAccumulator.of(
-                    sparkCombineFn.mergeCombiners(
-                        acc1.getOrDecode(iterAccumCoder), acc2.getOrDecode(iterAccumCoder)),
-                    iterAccumCoder));
+    JavaPairRDD<K, SerializableAccumulator<KV<K, AccumT>>> accumulatedResult;
+    if (partitioner == null) {
```

Comment:
I don't really like doing this null check and calling different functions `combineByKey` function based on whether the `partitioner` is present or not. I can change the `getPartitioner` function to always return a partitioner and then this way I don't have to do this check here. Thoughts @iemejia ?

Reply:
We should get rid of `nullable` case and have one translation only, see note above